lucille-core/src/test/java/com/kmwllc/lucille/stage/ApplyJSONNataTest.java

```diff
@@ -0,0 +1,73 @@
+package com.kmwllc.lucille.stage;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertThrows;
+import java.util.List;
+import org.junit.Test;
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.node.ObjectNode;
+import com.kmwllc.lucille.core.Document;
+import com.kmwllc.lucille.core.Stage;
+import com.kmwllc.lucille.core.StageException;
+
+public class ApplyJSONNataTest {
+
+  private final StageFactory factory = StageFactory.of(ApplyJSONNata.class);
+  private final ObjectMapper mapper = new ObjectMapper();
+
+  @Test 
+  public void testConfig() {
+    assertThrows(StageException.class, () -> factory.get("ApplyJSONNataTest/noExpression.conf"));
+  }
+
+  @Test 
+  public void testInvalidExpression() {
+    assertThrows(StageException.class, () -> factory.get("ApplyJSONNataTest/invalidExpression.conf"));
+  }
+
+  @Test
+  public void testApplyToFullDocument() throws StageException {
+    Stage invalidStage = factory.get("ApplyJSONNataTest/fullInvalid.conf");
+    Stage validStage = factory.get("ApplyJSONNataTest/fullValid.conf");
+
+    Document doc = Document.create("id");
+    doc.setField("foo", "bar");
+
+    invalidStage.processDocument(doc);
+    assertEquals(2, doc.getFieldNames().size());
+    assertEquals("id", doc.getId());
+    assertEquals("bar", doc.getString("foo"));
+
+    validStage.processDocument(doc);
+    assertEquals(2, doc.getFieldNames().size());
+    assertEquals("id", doc.getId());
+    assertEquals(List.of("id", "foo"), doc.getStringList("keys"));
+  }
+
+  @Test 
+  public void testApplyToField() throws StageException {
+    Stage withDest = factory.get("ApplyJSONNataTest/withDest.conf");
```

Comment:
minor point but "with" is often used as the prefix of a Builder method name
suggest renaming to stageWithDest and stageWithoutDest for clarity